xdsl/ir.py

```diff
@@ -15,12 +15,41 @@
 OpT = TypeVar('OpT', bound='Operation')
 
 
+@dataclass
+class Dialect:
+    """Contains the operations and attributes of a specific dialect"""
+    __operations: list[Operation] = field(default_factory=list,
```

Comment:
AFAIK, you should only use single underscores here.
Double underscores are only used for "special" methods from what I understand.

Reply:
Reading up on it, it seems that `__` gets name mangled, whereas `_` should only be used internally. Not sure where I read that `_` is by convention protected and `__` is by convention private. I change this back to `_`.